### ISSUE TITLE
import failing on editable install

### ISSUE THREAD

**Issue by marcofrasvda** [astral-sh/ty#3834]
### Summary

In VSCode, `ty` marks an error on a package coming from an editable install, even though the code works as expected.
I thought this was a regression since I've been working like this with ty for a while, the issue seems to exist in less recent versions of the ty extension though, so I'm not sure what's going on.
I can reproduce this both using Python 3.9 and 3.13.

EDIT: it seems this has something to do with the presence of `setup.py`, removing it fixes the issue, but since I cannot get rid of it right now and that it seems a false positive, I believe the report is still valid.

#### How to reproduce

Clone this somewhere in your computer:
```sh
git clone https://github.com/marcofrasvda/examplelib.git
```

Then update the path to the cloned `examplelib` in this script and run it:
```sh
rm -rf ty_repro/
mkdir ty_repro
cd ty_repro

python3.9 -m venv .venv
#python3.13 -m venv .venv
source .venv/bin/activate

pip install --editable path/to/examplelib

echo -e "from examplelib import system\n\nprint(system.is_cloud())\n" > main.py
code .
```

The output I see:
```sh
Obtaining file:///home/user/Public/examplelib
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Building wheels for collected packages: examplelib
  Building editable for examplelib (pyproject.toml) ... done
  Created wheel for examplelib: filename=examplelib-0.55.0-0.editable-py3-none-any.whl size=2751 sha256=1
c30e0afd714b13a0906d7507376b4608256481b85b269474961da5d5000ba27
  Stored in directory: /tmp/pip-ephem-wheel-cache-iavnbfbo/wheels/89/68/c9/d0ec1f942c749e1d8f34d21dec3e4a
ec3df16ed97755a188d7
Successfully built examplelib
Installing collected packages: examplelib
Successfully installed examplelib-0.55.0
```

Diagnostic when hovering `import examplelib`:
```
Cannot resolve imported module `examplelib`

info: Searched in the following paths during module resolution:
info:   1. /home/user/Public/ty_repro (first-party code)
info:   2. vendored://stdlib (stdlib typeshed stubs vendored by ty)
info:   3. /home/user/Public/ty_repro/.venv/lib/python3.13/site-packages (site-packages)
info:   4. /home/user/Public/ty_repro/.venv/lib64/python3.13/site-packages (site-packages)
info: make sure your Python environment is properly configured:
  https://docs.astral.sh/ty/modules/#python-environment
  ty(unresolved-import)
```

The environment seems to be set correctly though:
```sh
(.venv) user@pad:~/Public/ty_repro$ which python
/home/user/Public/ty_repro/.venv/bin/python
(.venv) user@pad:~/Public/ty_repro$ python --version
Python 3.13.5
(.venv) user@pad:~/Public/ty_repro$ python
Python 3.13.5 (main, May  5 2026, 21:05:52) [GCC 14.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
Ctrl click to launch VS Code Native REPL
>>> from examplelib import system
>>> print(system.is_cloud())
True
>>> 
```

#### Things that don't solve the issue

- reloading the VSCode window
- reloading the `ty` server
- selecting the Python interpreter via VSCode settings
- changing the interpreter used by `ty` via `pyproject` settings
- removing the shebang form `setup.py`

### Version

0.0.51

**Comment by MichaReiser** [comment]
This sounds a lot like https://github.com/astral-sh/ty/issues/475

**Comment by carljm** [comment]
Yes this is #475.

**Comment by marcofrasvda** [comment]
Workaround: `pip install --editable path/to/examplelib --config-settings editable_mode=compat`
